### ISSUE TITLE
Add support for cocotb TESTCASE specifier

### ISSUE THREAD

**Issue by tj-scherer** [pyuvm/pyuvm#152]
When running tests with the `@cocotb.test()` decorator, you are able to specify a TESTCASE in the call to `make`, this does not appear to work for `@pyuvm.test()`. The test (named `Test`) is located when TESTCASE is not specified.

```log
#      0.00ns INFO     cocotb                             Running on ModelSim SE-64 version 2021.3 2021.07
# 
#      0.00ns INFO     cocotb                             Running tests with cocotb v1.8.0.dev0 from C:\work\RioxI2cCtrl\tb\cocotb\.conda\lib\site-packages\cocotb
# 
#      0.00ns INFO     cocotb                             Seeding Python random module with 1689701124
# 
#      0.00ns ERROR    cocotb.regression                  Requested Test from module test_riox_i2c_ctrl isn't a cocotb.test decorated coroutine
# 
#      0.00ns ERROR    gpi                                Passing event to upper layer failed
```

```log
Traceback (most recent call last):
  File "C:\work\RioxI2cCtrl\tb\cocotb\.conda\lib\site-packages\cocotb\__init__.py", line 233, in _initialise_testbench
    _initialise_testbench_(argv_)
  File "C:\work\RioxI2cCtrl\tb\cocotb\.conda\lib\site-packages\cocotb\__init__.py", line 316, in _initialise_testbench_
    regression_manager = RegressionManager.from_discovery(top)
  File "C:\work\RioxI2cCtrl\tb\cocotb\.conda\lib\site-packages\cocotb\regression.py", line 174, in from_discovery
    return cls(dut, tests)
  File "C:\work\RioxI2cCtrl\tb\cocotb\.conda\lib\site-packages\cocotb\regression.py", line 153, in __init__
    for test in tests:
  File "C:\work\RioxI2cCtrl\tb\cocotb\.conda\lib\site-packages\cocotb\regression.py", line 227, in _discover_tests
    raise ImportError(
ImportError: Failed to find requested test Test
```

**Comment by raysalemi** [comment]
Yes.  Good point.  There was no way to make the TESTCASE option work easily with `@pyuvm.test()`

**Comment by raysalemi** [comment]
I don't know how to fix this.  @ktbarrett, may know, but we haven't focused on it as there is a workaround.

**Comment by ktbarrett** [comment]
This was fixed in #110.

**Comment by raysalemi** [comment]
> This was fixed in #110.

I think we decided not to fix it in #110 because of point 2.

**Comment by raysalemi** [comment]
RIght.  I chose not to teach that path since it is a bit complex.
